Source/WebCore/rendering/LegacyInlineBox.cpp

```diff
@@ -190,6 +192,23 @@ void LegacyInlineBox::adjustPosition(float dx, float dy)
         downcast<RenderBox>(renderer()).move(LayoutUnit(dx), LayoutUnit(dy));
 }
 
+void LegacyInlineBox::adjustBlockDirectionPosition(float delta, bool adjustStaticPosition)
+{
+    if (isHorizontal())
+        LegacyInlineBox::adjustPosition(0, delta);
+    else
+        LegacyInlineBox::adjustPosition(delta, 0);
+
+    if (adjustStaticPosition && renderer().isOutOfFlowPositioned()) {
+        RenderBox& box = static_cast<RenderBox&>(renderer());
+        if (box.style().hasStaticBlockPosition(isHorizontal())) {
+            ASSERT(box.layer());
+            box.layer()->setStaticBlockPosition(box.layer()->staticBlockPosition() + LayoutUnit(delta));
```

Comment:
`box->checkedLayer()-> setStaticBlockPosition()` would be safer since it would guarantee the RenderLayer stays alive during the duration of the setStaticBlockPosition() call.